addons/account/models/sequence_mixin.py

```diff
@@ -60,15 +60,20 @@ def _deduce_sequence_number_reset(self, name):
         """
         def _check_grouping(grouping, optional=None, required=None):
             sequence_dict = sequence.groupdict()
-            return all(key in sequence_dict for key in (optional or [])) and all(sequence_dict.get(key) for key in (required or []))
+            check = all(key in sequence_dict for key in (optional or [])) and all(sequence_dict.get(key) for key in (required or []))
+            if 'year' in (required or []):
+                check &= 2000 <= int(sequence_dict.get('year', -1)) <= 2100 or (len(sequence_dict.get('year', '')) == 2 and bool(sequence_dict.get('prefix2')))
```

Comment:
This adds the constraint of having `prefix2` (exactly, and right after `year`) in overridden regex's if we want to handle 2 digits years. I'm not sure how to avoid that without making it too complicated.